_posts/2020-05-11-BART-Summary.md

```diff
@@ -0,0 +1,73 @@
+
+**Token Masking** random tokens are sampled and replaced with [MASK]<br />
+**Token Deletion** similar to masking but the token sampled are deleted and the model has to add new token in their place.<br />
+**Token Infilling** a number of text spans, i.e. contiguous group tokens, are sampled, and then they are replaced by the [MASK] token.<br />
+**Sentence Permutation** random shuffling of the document’s sentences.<br />
+**Document Rotation** a token is chosen randomly to be the start of the document, the section before the starting token is appended at the end.<br />
+
+Intuitively the techniques that work at the sentence level should help the LM learn the different roles of sentences in a paragraph or longer text and so help in NLG tasks.
+
+Besides the pre-training techniques, the authors also compare different LM objectives focusing on the ones used by BERT and GPT and also the ones that tried to get the best of both worlds.
+
+**Autoregressive, left to right, LM** (GPT-2)<br />
+**Masked LM**(BERT) replace 15% of the token with [MASK] and predict the corresponding words.<br />
+**Permuted LM** (XLNet) left to right, autoregressive LM training but with the order of the words to predict chosen at random. (worth looking into more, very interesting approach)<br />
+**Multitask Masked LM** (UniLM) combination of right-to-left, left-to-right and Bidirectional. ⅓ of the time using each with shared parameters.<br />
+**Masked Seq2Seq** (MASS) masking a span containing 50% of the tokens and train to  predict the masked tokens.<br />
+        
+### **Results of the first experiment**
+![](https://lh6.googleusercontent.com/YxPMuTJc7EY2rFYIXUBVIYMjV-rloyEj2UmgJ6pbxyyMDCWzdwu4KOgRErOKJcmDe4QfC7LO-2bGE6-_0pCF1lRwJfFbjGvBbuk73oFQ8AgMdHAYDNIwDH8HlEcBI15SQKTUMIUhc_8)
+    
+From the results of this first experiments the authors draw some important conclusions.<br />
+**Token masking is crucial** Only the configurations with token masking or its variations achieve consistently great performance on different tasks.<br />
+**Left-to-right pre-training improves NLG** The Classical Language Model objective despite not doing well in inference or QA tasks achieves SOTA on ELI5(Explain Like I'm 5) a NLG benchmark that with longer sequences values coherent and grammatically correct text.<br />
+**Bidirectional encoders are crucial for QA** Ignoring future context hinders the performance of left-to-right models.<br />
+
+One Interesting note from the authors in this experiment explains how while pre-training techniques and LM objectives are important they don't tell the whole story. They report that their Permuted Language model performs much worse than XLNet due to some of the valuable architectural innovations introduced in XLNet that are not present in BART.
```

Comment:
This sentence: "One Interesting note from the authors in this experiment explains how while pre-training techniques and LM objectives are important they don't tell the whole story" is a bit cumbersome to read. I would suggest restructuring a bit. 

Suggestion: "While pre-training techniques and LM objectives are important, the authors make note of the fact that they do not provide the full picture."